test/es-module/test-esm-loader-mock.mjs

```diff
@@ -1,9 +1,73 @@
-// Flags: --loader ./test/fixtures/es-module-loaders/mock-loader.mjs
+// Flags: --no-warnings
```

Comment:
nit: it would be more interesting to pass a `--import` like users would typically do
```suggestion
// Flags: --import ./test/fixtures/es-module-loaders/mock-register.mjs
```

Reply:
This would be a more significant refactor of the test, though. I did refactor to use `--import` for the named exports test in the other PR. We could also land this like this and refactor further in the other PR.

Reply:
It depends what do we want this PR to be. It would be helpful to be able to point folks to this PR as a showcase of how to migrate from `--loader`/`globalPreload` to `--import`/`initialize`. If we split this into two PRs, it will be harder to follow, it wouldn’t make much sense to do a follow up PR. If we don’t care about showcasing that (because it’s done elsewhere, or we’re confident the docs are good enough), sure let’s keep it as is.

Reply:
This PR isn't the best example of a migration because the "before" case is so atypical. The named exports or resolve type tests that I migrated in the other PR are good candidates though. We could move those migrations into one or two standalone PRs if you think those would be good to point to.

Reply:
I don't think that use case is as uncommon as you think (at least among the users of `globalPreload`); the good thing about it is that it touches almost all the corners of the ~loaders~ module customization hooks API, so IMO it does make a good example for power users. That being said those power users probably don't need us to tell them what belong to the `--import` module, so I'm fine with whatever, feel free to ignore.

Reply:
The use case for this one is common but the implementation before my refactoring is very unusual. If the goal is to show how to migrate from `globalPreload`, I think a simpler and more typical example would be easier to understand.